### PR TITLE
Add a simple gauge metrics for memory used.

### PR DESCRIPTION
**Pull request by PremyslCerny**

### CODE REVIEW

**Comment by PremyslCerny** [comment]
@ateska Please see. Any suggestions are welcome.

File: asab/metrics/service.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import configparser
 import logging
 import asyncio
+import os
+import psutil
```
Comment (ateska) [inline asab/metrics/service.py:5]: This is the external dependency. I would like to try it without it - you can basically read that from `/proc/self` file system AFAIK.

**Review by ateska — CHANGES_REQUESTED** [review]

File: asab/metrics/service.py [hunk 2]
```diff
@@ -63,6 +65,20 @@ def __init__(self, app, service_name):
 		self.MemstorTarget = MetricsMemstorTarget(self, 'asab:metrics:memory')
 		self.Targets.append(self.MemstorTarget)
 
+		# Create native metrics
+		self.Process = psutil.Process(os.getpid())
+
+		self.MemoryGauge = self.create_gauge(
+			"asab.memory",
+			init_values={
+				"memory.used": self.Process.memory_info().rss,
```
Comment (ateska) [inline asab/metrics/service.py:74]: The metric name should be consistent with Linux memory/swap naming.

**Review by ateska — CHANGES_REQUESTED** [review]

File: asab/metrics/service.py [hunk 3]
```diff
@@ -63,6 +64,18 @@ def __init__(self, app, service_name):
 		self.MemstorTarget = MetricsMemstorTarget(self, 'asab:metrics:memory')
 		self.Targets.append(self.MemstorTarget)
 
+		# Create native metrics
+		self.ProcessId = os.getpid()
+
+		self.MemoryGauge = self.create_gauge(
+			"asab.memory",
```
Comment (ateska) [inline asab/metrics/service.py:71]: Status information about the OS process.

File: asab/metrics/service.py [hunk 4]
```diff
@@ -63,6 +64,18 @@ def __init__(self, app, service_name):
...
+			init_values=self._get_process_memory_info(),
+			tags={
+				"unit": "bytes",
+				"help": "Memory consumed by the process.",
```
Comment (ateska) [inline asab/metrics/service.py:75]: This is not only a memory consumption ...

**Review by ateska — CHANGES_REQUESTED** [review]

**Review by pagan-coder — APPROVED** [review]

File: asab/metrics/service.py [hunk 5]
```diff
@@ -63,6 +64,17 @@ def __init__(self, app, service_name):
 		self.MemstorTarget = MetricsMemstorTarget(self, 'asab:metrics:memory')
 		self.Targets.append(self.MemstorTarget)
 
+		# Create native metrics
+		self.ProcessId = os.getpid()
+
+		self.MemoryGauge = self.create_gauge(
+			"asab.process",
```
Comment (ateska) [inline asab/metrics/service.py:71]: Don't say asab, please. It is `os.stat` for example.

**Review by ateska — CHANGES_REQUESTED** [review]

**Comment by ateska** [comment]
Can I see the output of this metrics, please? 

:-)

**Comment by PremyslCerny** [comment]
@ateska Of course, they are key value pairs:

```
VmPeak 184795136
VmSize 118030336
VmLck 0
VmPin 0
VmHWM 29437952
VmRSS 29437952
VmData 27840512
VmStk 135168
VmExe 2707456
VmLib 5586944
VmPTE 114688
VmSwap 0
```

**Comment by ateska** [comment]
Nice.

**Pull request merged**